kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -498,8 +517,19 @@ private void writeRecord(SinkRecord record) {
       endOffsets.put(currentEncodedPartition, 0L);
     }
 
-    RecordWriter writer = getWriter(record, currentEncodedPartition);
-    writer.write(record);
+    RecordWriter writer = getWriter(currentEncodedPartition);
+    try {
+      writer.write(record);
+    } catch (DataException e) {
+      if (reporter != null) {
+        reporter.report(record, e);
+        log.warn("Errant record written to DLQ due to: {}", e.getMessage());
+        writer.close();
+        writers.remove(currentEncodedPartition); // writer keeps exception state otherwise
```

Comment:
the comment doesn't read very well for me. 
Does this work for cases where the record is both first or somewhere in the middle of the currentEncodedPartiion?

Reply:
Updated, this is N/A anymore.